packages/trader/src/App/Components/Elements/market-countdown-timer.jsx

```diff
@@ -0,0 +1,212 @@
+                    return whenMarketOpens(days_offset + 1, target_symbol);
+                }
+                const date_str = target_date.toISOString().substring(0, 11);
+                const getUTCDate = hour => new Date(`${date_str}${hour}Z`);
+                for (let i = 0; i < open?.length; i++) {
+                    const diff = +getUTCDate(open[i]) - +new Date();
+                    if (diff > 0) {
+                        remaining_time_to_open = +getUTCDate(open[i]);
+                        if (isMounted() && target_symbol === symbol) {
+                            return setWhenMarketOpens({ days_offset, opening_time: open[i], remaining_time_to_open });
+                        }
+                    }
+                }
+                whenMarketOpens(days_offset + 1, target_symbol);
+            }
+        };
+
+        whenMarketOpens(0, symbol);
+    }, [isMounted, symbol]);
+
+    React.useEffect(() => {
+        let timer;
+        if (when_market_opens?.remaining_time_to_open) {
+            timer = setTimeout(async () => {
+                setTimeLeft(calculateTimeLeft(when_market_opens?.remaining_time_to_open));
+                if (when_market_opens.remaining_time_to_open <= 1 && typeof setActiveSymbols === 'function')
+                    await setActiveSymbols();
+            }, 1000);
+        }
+        return () => {
+            if (timer) {
+                clearTimeout(timer);
+            }
+        };
+    }, [time_left, when_market_opens]);
+
+    let timer_components = '';
+
+    Object.keys(time_left).forEach(interval => {
+        if (interval === 'days') {
+            if (time_left.days) {
+                timer_components += `${time_left.days} ${localize(time_left.days > 1 ? 'days' : 'day')}`;
+            }
+        } else if (time_left[interval] !== 0) {
+            const value = time_left[interval];
+            timer_components +=
+                interval === 'hours'
+                    ? `${time_left.days ? ', ' : ''}${value} ${localize(time_left.hours > 1 ? 'hrs' : 'hr')}`
+                    : ` ${localize('and')} ${value} ${localize(time_left.minutes > 1 ? 'mins' : 'min')}`;
+        }
+    });
+
+    const { opening_time, days_offset } = when_market_opens;
+    let opening_time_banner = null;
+    if (opening_time) {
+        const target_date = moment(new Date()).add(days_offset, 'days');
+        const opening_time_moment_obj = moment(opening_time, 'HH:mm');
+        const opening_time_hour = opening_time_moment_obj.format('HH');
+        const opening_time_min = opening_time_moment_obj.format('mm');
+        const formatted_opening_time =
+            Number(opening_time_hour) > 11
+                ? `${Number(opening_time_hour) % 12}:${opening_time_min} pm`
+                : `${Number(opening_time_hour)}:${opening_time_min} am`;
+        const opening_date = target_date.format('D MMM YYYY');
+        const opening_day = target_date.format('dddd');
+        opening_time_banner = (
```

Comment:
You could create this during rendering rather than here, no need to assign it to memory when the below condition is met (line 158).